Index: package.json
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>{\r\n  \"name\": \"sql-databases\",\r\n  \"version\": \"1.0.0\",\r\n  \"description\": \"\",\r\n  \"main\": \"index.js\",\r\n  \"scripts\": {\r\n    \"test\": \"echo \\\"Error: no test specified\\\" && exit 1\",\r\n    \"dev\": \"nodemon server.js\"\r\n  },\r\n  \"keywords\": [],\r\n  \"author\": \"\",\r\n  \"license\": \"ISC\",\r\n  \"dependencies\": {\r\n    \"bcrypt\": \"^5.0.1\",\r\n    \"connect-mongo\": \"^4.6.0\",\r\n    \"cookie-parser\": \"^1.4.6\",\r\n    \"dotenv\": \"^16.0.2\",\r\n    \"express\": \"^4.18.1\",\r\n    \"express-handlebars\": \"^6.0.6\",\r\n    \"express-session\": \"^1.17.3\",\r\n    \"faker\": \"^5.5.3\",\r\n    \"knex\": \"^2.2.0\",\r\n    \"mongoose\": \"^6.6.0\",\r\n    \"normalizr\": \"^3.6.2\",\r\n    \"passport\": \"^0.6.0\",\r\n    \"passport-local\": \"^1.0.0\",\r\n    \"socket.io\": \"^4.5.1\",\r\n    \"sqlite3\": \"^5.0.11\",\r\n    \"yargs\": \"^17.6.0\"\r\n  },\r\n  \"devDependencies\": {\r\n    \"nodemon\": \"^2.0.19\"\r\n  }\r\n}\r\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/package.json b/package.json
--- a/package.json	(revision 6d522cbec823ad0eae1920a0186ac622b73c296e)
+++ b/package.json	(date 1665066693325)
@@ -1,5 +1,5 @@
 {
-  "name": "sql-databases",
+  "name": "process",
   "version": "1.0.0",
   "description": "",
   "main": "index.js",
@@ -19,13 +19,11 @@
     "express-handlebars": "^6.0.6",
     "express-session": "^1.17.3",
     "faker": "^5.5.3",
-    "knex": "^2.2.0",
     "mongoose": "^6.6.0",
     "normalizr": "^3.6.2",
     "passport": "^0.6.0",
     "passport-local": "^1.0.0",
     "socket.io": "^4.5.1",
-    "sqlite3": "^5.0.11",
     "yargs": "^17.6.0"
   },
   "devDependencies": {
Index: controllers/APIController.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>const numberGenerator = require('../utils/numberGenerator')\r\n\r\nconst generateRandoms = (req, res) => {\r\n    const result = numberGenerator(req.query.qty || 10e8)\r\n    return res.send(result)\r\n}\r\n\r\nmodule.exports = {\r\n    generateRandoms\r\n}
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/controllers/APIController.js b/controllers/APIController.js
--- a/controllers/APIController.js	(revision 6d522cbec823ad0eae1920a0186ac622b73c296e)
+++ b/controllers/APIController.js	(date 1665067981386)
@@ -1,8 +1,14 @@
 const numberGenerator = require('../utils/numberGenerator')
+const { fork } = require('child_process')
 
 const generateRandoms = (req, res) => {
-    const result = numberGenerator(req.query.qty || 10e8)
-    return res.send(result)
+    const child = fork('../utils/numberGenerator')
+    child.send(req.query.qty || 10e8)
+    child.on('message', result => {
+        return res.send(result)
+    })
+    //const result = numberGenerator()
+    //return res.send(result)
 }
 
 module.exports = {
Index: utils/numberGenerator.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>const numberGenerator = (qty) => {\r\n    const numbers = {}\r\n    for (let i = 0; i < qty; i++) {\r\n        let result = Math.ceil(Math.random() * 1000)\r\n        if (!numbers[result]) {\r\n            numbers[result] = 1\r\n        } else {\r\n            numbers[result]++\r\n        }\r\n    }\r\n    return numbers\r\n}\r\n\r\nmodule.exports = numberGenerator
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/utils/numberGenerator.js b/utils/numberGenerator.js
--- a/utils/numberGenerator.js	(revision 6d522cbec823ad0eae1920a0186ac622b73c296e)
+++ b/utils/numberGenerator.js	(date 1665067602607)
@@ -11,4 +11,9 @@
     return numbers
 }
 
+process.on('message', (qty) => {
+    const result = numberGenerator(qty)
+    process.send(result)
+})
+
 module.exports = numberGenerator
\ No newline at end of file
Index: .idea/.gitignore
===================================================================
diff --git a/.idea/.gitignore b/.idea/.gitignore
deleted file mode 100644
--- a/.idea/.gitignore	(revision 6d522cbec823ad0eae1920a0186ac622b73c296e)
+++ /dev/null	(revision 6d522cbec823ad0eae1920a0186ac622b73c296e)
@@ -1,5 +0,0 @@
-# Default ignored files
-/shelf/
-/workspace.xml
-# Editor-based HTTP Client requests
-/httpRequests/
Index: .idea/10_authentication.iml
===================================================================
diff --git a/.idea/10_authentication.iml b/.idea/10_authentication.iml
deleted file mode 100644
--- a/.idea/10_authentication.iml	(revision 6d522cbec823ad0eae1920a0186ac622b73c296e)
+++ /dev/null	(revision 6d522cbec823ad0eae1920a0186ac622b73c296e)
@@ -1,12 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
-<module type="WEB_MODULE" version="4">
-  <component name="NewModuleRootManager">
-    <content url="file://$MODULE_DIR$">
-      <excludeFolder url="file://$MODULE_DIR$/temp" />
-      <excludeFolder url="file://$MODULE_DIR$/.tmp" />
-      <excludeFolder url="file://$MODULE_DIR$/tmp" />
-    </content>
-    <orderEntry type="inheritedJdk" />
-    <orderEntry type="sourceFolder" forTests="false" />
-  </component>
-</module>
\ No newline at end of file
Index: .idea/modules.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+><?xml version=\"1.0\" encoding=\"UTF-8\"?>\r\n<project version=\"4\">\r\n  <component name=\"ProjectModuleManager\">\r\n    <modules>\r\n      <module fileurl=\"file://$PROJECT_DIR$/.idea/10_authentication.iml\" filepath=\"$PROJECT_DIR$/.idea/10_authentication.iml\" />\r\n    </modules>\r\n  </component>\r\n</project>
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.idea/modules.xml b/.idea/modules.xml
--- a/.idea/modules.xml	(revision 6d522cbec823ad0eae1920a0186ac622b73c296e)
+++ b/.idea/modules.xml	(date 1665066654014)
@@ -2,7 +2,7 @@
 <project version="4">
   <component name="ProjectModuleManager">
     <modules>
-      <module fileurl="file://$PROJECT_DIR$/.idea/10_authentication.iml" filepath="$PROJECT_DIR$/.idea/10_authentication.iml" />
+      <module fileurl="file://$PROJECT_DIR$/.idea/12_process.iml" filepath="$PROJECT_DIR$/.idea/12_process.iml" />
     </modules>
   </component>
 </project>
\ No newline at end of file
